### PR TITLE
Databases collection

### PR DESCRIPTION
**Pull request by lunika**
## Purpose

We want to define mongo IPs at the application level. In some project other applications can also use Mongo with a different version or server IP. A new variable is defined for each applications using mongo and its default value is `endpoint_mongodb_ips` variable value.

## Proposal

- [x] add `learninglocker_endpoint_mongodb_ips` variable
- [x] add `forum_endpoint_mongodb_ips` variable
- [x] add `edxapp_endpoint_mongodb_ips` variable
- [x] bump to version 4.3.0

### CODE REVIEW

File: apps/edxapp/vars/all/main.yml [hunk 1]
```diff
@@ -87,6 +87,7 @@ edxapp_mongodb_host: "edxapp-mongodb"
 edxapp_mongodb_replicaset: ""
 edxapp_mongodb_read_preference: ""
 edxapp_mongodb_secret_name: "edxapp-mongo-{{ edxapp_vault_checksum | default('undefined_edxapp_vault_checksum') }}"
+edxapp_endpoint_mongodb_ips: "{{ endpoint_mongodb_ips }}"
```
Comment (jmaupetit) [inline apps/edxapp/vars/all/main.yml:90]:
```suggestion
edxapp_endpoint_mongodb_ips: {{ endpoint_mongodb_ips }}
```

File: apps/forum/vars/all/main.yml [hunk 2]
```diff
@@ -25,3 +25,4 @@ forum_mongodb_host: "forum-mongodb"
 forum_mongodb_replicaset: ""
 forum_mongodb_read_preference: ""
 forum_mongodb_secret_name: "forum-mongo-{{ forum_vault_checksum | default('undefined_forum_vault_checksum') }}"
+forum_endpoint_mongodb_ips: "{{ endpoint_mongodb_ips }}"
```
Comment (jmaupetit) [inline apps/forum/vars/all/main.yml:28]:
```suggestion
forum_endpoint_mongodb_ips: {{ endpoint_mongodb_ips }}
```

File: apps/learninglocker/templates/services/app/macros/mongodb_uri.yml.j2 [hunk 3]
```diff
@@ -1,13 +1,13 @@
 {# compute the mongodb URI #}
-{%- macro mongodb_uri(env_type, trashable_env_types, learninglocker_mongodb_host, deployment_stamp, endpoint_mongodb_ips, learninglocker_mongodb_replicaset, learninglocker_mongodb_read_preference) -%}
+{%- macro mongodb_uri(env_type, trashable_env_types, learninglocker_mongodb_host, deployment_stamp, learninglocker_learninglocker_endpoint_mongodb_ips, learninglocker_mongodb_replicaset, learninglocker_mongodb_read_preference) -%}
```
Comment (jmaupetit) [inline apps/learninglocker/templates/services/app/macros/mongodb_uri.yml.j2]:
```suggestion
{%- macro mongodb_uri(env_type, trashable_env_types, learninglocker_mongodb_host, deployment_stamp, learninglocker_endpoint_mongodb_ips, learninglocker_mongodb_replicaset, learninglocker_mongodb_read_preference) -%}
```

File: apps/learninglocker/templates/services/app/macros/mongodb_uri.yml.j2 [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
     {%- if env_type in trashable_env_types -%}
         mongodb://$(MONGODB_USER):$(MONGODB_PASSWORD)@{{ learninglocker_mongodb_host }}-{{ deployment_stamp }}/$(MONGODB_DATABASE)
     {%- else -%}
         {%- set mondogdb_service_name = "learninglocker-mongodb-%s" | format(deployment_stamp) -%}
         {%- set ns = namespace(services="") -%}
-        {%- for ip in endpoint_mongodb_ips -%}
+        {%- for ip in learninglocker_learninglocker_endpoint_mongodb_ips -%}
```
Comment (jmaupetit) [inline apps/learninglocker/templates/services/app/macros/mongodb_uri.yml.j2]:
```suggestion
        {%- for ip in learninglocker_endpoint_mongodb_ips -%}
```

File: apps/learninglocker/templates/services/app/macros/mongodb_uri.yml.j2 [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
             {%- set ns.services = "%s%s:%s" | format(ns.services, mondogdb_service_name, learninglocker_mongodb_port | int + loop.index0) -%}
-            {%- if loop.index < endpoint_mongodb_ips | length -%}
+            {%- if loop.index < learninglocker_learninglocker_endpoint_mongodb_ips | length -%}
```
Comment (jmaupetit) [inline apps/learninglocker/templates/services/app/macros/mongodb_uri.yml.j2]:
```suggestion
            {%- if loop.index < learninglocker_endpoint_mongodb_ips | length -%}
```

**Review by sampaccoud — APPROVED** [review]

File: apps/learninglocker/vars/all/main.yml [hunk 6]
```diff
@@ -30,6 +30,7 @@ learninglocker_mongodb_host: "learninglocker-mongodb"
 learninglocker_mongodb_replicaset: ""
 learninglocker_mongodb_read_preference: ""
 learninglocker_mongodb_secret_name: "learninglocker-mongodb-{{ learninglocker_vault_checksum | default('undefined_learninglocker_vault_checksum') }}"
+learninglocker_endpoint_mongodb_ips: "{{ endpoint_mongodb_ips }}"
```
Comment (jmaupetit) [inline apps/learninglocker/vars/all/main.yml:33]:
```suggestion
learninglocker_endpoint_mongodb_ips: {{ endpoint_mongodb_ips }}
```

**Review by jmaupetit — COMMENTED** [review]
I think you should not cast `endpoint_mongodb_ips` to a string as it's a list

File: apps/edxapp/vars/all/main.yml (hunk 1 above)
Comment (lunika) [reply apps/edxapp/vars/all/main.yml:90]: No it doesn't work, I test it

File: apps/learninglocker/templates/services/app/macros/mongodb_uri.yml.j2 (hunk 3 above)
Comment (lunika) [reply apps/learninglocker/templates/services/app/macros/mongodb_uri.yml.j2]: Holly sh**

File: apps/edxapp/vars/all/main.yml (hunk 1 above)
Comment (jmaupetit) [reply apps/edxapp/vars/all/main.yml:90]: If this entry is casted as a list despite quotes forget it then!

**Review by jmaupetit — APPROVED** [review]

**Pull request merged**